apps/integration/integration/automation/integrate

```diff
@@ -0,0 +1,202 @@
+      TAG="$2"
+      echo "Using tag ${TAG}"
+      shift
+      shift
+      ;;
+  esac
+done
+
+NOW="$(date +"%Y-%m-%d_%H_%M_%S")"
+ARCH="amd64"
+[[ "$(uname -m)" = "arm64" ]] && ARCH="arm64"
+
+RUNDIR="apps/integration/runs/${NOW}"
+GIT_LOGFILE="${RUNDIR}/git.log"
+DOCKER_LOGFILE="${RUNDIR}/docker.log"
+POETRY_LOGFILE="${RUNDIR}/poetry.log"
+PYTEST_LOGFILE="${RUNDIR}/pytest.log"
+QUERY_LOGFILE="${RUNDIR}/test_queries.log"
+
+main() {
+  [[ ! -d ".git" ]] && die "Please run this script from sycamore root!"
+  mkdir -p "${RUNDIR}"
+  echo "Building/testing tag ${TAG}" >&2
+  echo "Get the newest git commits" >&2
+  if checkout_main_if_new; then
+    echo "Changes detected. Running Tests" >&2
+    poetry install --no-root > "${POETRY_LOGFILE}" 2>&1 \
+        && { [[ $SKIP_BUILD ]] || build_images > "${DOCKER_LOGFILE}" 2>&1; } \
+        && { [[ $SKIP_TESTS ]] || runtests > "${PYTEST_LOGFILE}" 2>&1; }
+    local passed_tests=$?
+    [[ $SKIP_PUSH ]] || push_images >> "${DOCKER_LOGFILE}" 2>&1
+    handle_outputs $passed_tests
+  else
+    echo "No changes detected. Skipping integration tests" >&2
+  fi
+}
+
+error() {
+  echo "ERROR: $@" >&2
+}
+
+die() {
+  error "$1"
+  exit 1
+}
+
+checkout_main_if_new() {
+  old_sha="$(git rev-parse HEAD)"
+  git fetch origin main > "${GIT_LOGFILE}" 2>&1
+  new_sha="$(git rev-parse FETCH_HEAD)"
+  if [[ "${old_sha}" != "${new_sha}" ]]; then
+    [[ -z $(git status --porcelain) ]] \
+      || { echo "Working tree not clean" > "${GIT_LOGFILE}" && return 1; }
+    git pull --rebase origin main >> "${GIT_LOGFILE}" 2>&1
+    return 0
+  else
+    return 1
+  fi
+}
+
+build_images() {
+  echo "Building all images" >&2
+  docker-build-hub apps/crawler/crawler/http/Dockerfile \
+  && docker-build-hub apps/crawler/crawler/s3/Dockerfile \
+  && docker-build-hub apps/importer/Dockerfile.buildx \
+  && docker-build-hub apps/opensearch/Dockerfile \
+  && docker-build-hub apps/jupyter/Dockerfile.buildx --build-arg=TAG="${TAG}" \
+  && docker-build-hub apps/demo-ui/Dockerfile.buildx \
+  && docker-build-hub apps/remote-processor-service/Dockerfile.buildx \
+  && return 0
+  return 1
+}
+
+handle_outputs() {
+  echo "Handling test outputs" >&2
+  local passed_tests="$1"
+  mv test-output.log "${QUERY_LOGFILE}"
+  if [[ ${passed_tests} = 0 ]]; then
+    touch "${RUNDIR}/passed"
+  else
+    touch "${RUNDIR}/failed"
+  fi
+  aws s3 cp -r "${RUNDIR}" "s3://sycamore-ci/${ARCH}"
+}
+
+push_images() {
+  echo "Pushing tested images to dockerhub" >&2
+  docker-push-hub apps/crawler/crawler/http/Dockerfile \
+  && docker-push-hub apps/crawler/crawler/s3/Dockerfile \
+  && docker-push-hub apps/importer/Dockerfile.buildx \
+  && docker-push-hub apps/opensearch/Dockerfile \
+  && docker-push-hub apps/jupyter/Dockerfile.buildx \
+  && docker-push-hub apps/demo-ui/Dockerfile.buildx \
+  && docker-push-hub apps/remote-processor-service/Dockerfile.buildx \
+  && return 0
+  return 1
+}
+
+runtests() {
+  docker system prune -f --volumes
```

Comment:
This is a pretty big hammer; my guess is the purpose is to clean up integration test volumes, but it will also clean up build caches and other people's stuff.
I'd suggest:
```
docker volume rm integration_crawl_data integration_jupyter_data integration_opensearch_data
docker compose -p integration up reset
```

That will clean up integration specific volumes but leave everything else.
Optionally clean them after successful testing and verify there are no volumes named `*integration*`

Reply:
There's not, as far as I can tell, a good way to get testcontainers to do project names...